package xin.vanilla.aotake.mixin;

import net.minecraft.server.level.ServerPlayer;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import xin.vanilla.aotake.data.world.ChunkVaultSession;

@Mixin(ServerPlayer.class)
public abstract class ServerPlayerMixin {

    @Inject(method = "closeContainer", at = @At("TAIL"))
    private void aotake$afterCloseContainer(CallbackInfo ci) {
        ChunkVaultSession.onPlayerCloseContainer((ServerPlayer) (Object) this);
    }
}
